modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/service/impl/DDMFormInstanceLocalServiceImpl.java

```diff
@@ -670,11 +681,21 @@ protected void updateWorkflowDefinitionLink(
 			ServiceContext serviceContext)
 		throws PortalException {
 
-		String workflowDefinition = getWorkflowDefinition(
+		String workflowDefinitionName = getWorkflowDefinition(
 			settingsDDMFormValues);
 
-		if (workflowDefinition.equals("no-workflow")) {
-			workflowDefinition = "";
+		String workflowDefinition = "";
+
+		if (Validator.isNotNull(workflowDefinitionName) &&
+			!workflowDefinitionName.equals("no-workflow")) {
+
+			KaleoDefinition kaleoDefinition =
+				_kaleoDefinitionLocalService.getKaleoDefinition(
+					workflowDefinitionName, serviceContext);
+
+			workflowDefinition =
+				workflowDefinitionName + StringPool.AT +
+					kaleoDefinition.getVersion();
```

Comment:
You did a good job here, @rodrigopaulino. 😄 

I just have a suggestion regarding the naming to keep consistency, please check it below:

- Rename `workflowDefinitionName` to `workflowDefinition`
- Rename `workflowDefinition` to `latestWorkflowDefinition` or `currentWorkflowDefinition`

Brian loves consistency, therefore if there is a method named `getWorkflowDefinition` which returns a value to a variable named `workflowDefinitionName` then I believe he'll request a change.

Reply:
Sure thing, @natocesarrego.